packages/js/e2e-utils/src/flows/with-rest-api.js

```diff
@@ -300,5 +300,35 @@ export const withRestApi = {
 		} else {
 			return;
 		}
-	}
+	},
+	/**
+	 * Create a product category and return the ID. If the category already exists, the ID of the existing category is returned.
+	 *
+	 * @param {String} categoryName The name of the category to create
+	 * @returns {Promise<Number>} The ID of the category.
+	 */
+	createProductCategory: async ( categoryName ) => {
+		const payload = { name: categoryName };
+		let categoryId;
+
+		try {
```

Comment:
The `try/catch` is needed here because it uses `expect` to test the success of the api call. The logic could be updated to check using an `if` condition.

It would be a cleaner implementation if the first api call was a [search](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-product-categories) to see if the category already exists. Then, if not, create it. On the smoke test sites, the category will exist on almost every test run.

Reply:
That's a great call out to use the search! I ended up converting the category name to a slug, then using the `slug` parameter to perform the search: https://github.com/woocommerce/woocommerce/pull/31055/files#diff-c2f9c0776ea905986d1e12c692703da2307805845e08ec422865c6fefc421ae5R317